### PR TITLE
Dell,Arrow : Migrating test cases in dell,arrow packages to junit5

### PR DESCRIPTION
**Pull request by DaVincii**
This pull request is related to the issue "Move JUnit4 tests to JUnit5 #7160 ".

### CODE REVIEW

File: dell/src/test/java/org/apache/iceberg/dell/mock/ecs/MockS3Client.java [hunk 1]
```diff
@@ -93,6 +93,8 @@
 import org.apache.iceberg.relocated.com.google.common.io.ByteStreams;
 import org.junit.Assert;
```
Comment (nastra) [inline dell/src/test/java/org/apache/iceberg/dell/mock/ecs/MockS3Client.java:94]: this is still using Junit4's `Assert`

File: dell/src/test/java/org/apache/iceberg/dell/ecs/TestPropertiesSerDesUtil.java [hunk 2]
```diff
@@ -20,8 +20,8 @@
 
 import java.util.Map;
 import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
-import org.junit.Assert;
-import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
```
Comment (nastra) [inline dell/src/test/java/org/apache/iceberg/dell/ecs/TestPropertiesSerDesUtil.java:23]: please use AssertJ assertions. See also https://iceberg.apache.org/contribute/#assertj

File: dell/src/test/java/org/apache/iceberg/dell/ecs/TestEcsSeekableInputStream.java [hunk 3]
```diff
@@ -27,6 +27,8 @@
 import org.junit.ClassRule;
 import org.junit.Test;
```
Comment (nastra) [inline dell/src/test/java/org/apache/iceberg/dell/ecs/TestEcsSeekableInputStream.java:28]: this is still using Junit4 tests and hasn't been moved to Junit5

**Review by nastra — CHANGES_REQUESTED** [review]
`iceberg-dell` and `iceberg-arrow` projects needs to be configured in `build.gradle` to run with Junit5 via 
```
 test {
    useJUnitPlatform()
  }
```

Also formatting seems to be off in some places (please run `./gradlew spotlessApply` and the conversion from Junit4 to Junit5 hasn't been fully completed in the touched test files

File: dell/src/test/java/org/apache/iceberg/dell/ecs/TestPropertiesSerDesUtil.java (hunk 2 above)
Comment (DaVincii) [reply dell/src/test/java/org/apache/iceberg/dell/ecs/TestPropertiesSerDesUtil.java]: This class uses @classRule and can't be migrated without some significant refactoring which I feel need some more experience to do. Should I split this pull request into arrow and dell packages as arrow doesn't have any classRules?

File: dell/src/test/java/org/apache/iceberg/dell/mock/ecs/MockS3Client.java (hunk 1 above)
Comment (DaVincii) [reply dell/src/test/java/org/apache/iceberg/dell/mock/ecs/MockS3Client.java]: This was unused import, removed it.

File: dell/src/test/java/org/apache/iceberg/dell/ecs/TestEcsSeekableInputStream.java (hunk 3 above)
Comment (DaVincii) [reply dell/src/test/java/org/apache/iceberg/dell/ecs/TestEcsSeekableInputStream.java:29]: Same as above, this one too uses ClassRule.

File: dell/src/test/java/org/apache/iceberg/dell/mock/ecs/EcsS3MockRule.java [hunk 4]
```diff
@@ -71,7 +72,7 @@ public static EcsS3MockRule manualCreateBucket() {
   /** Load rule from thread local and check bucket */
   public static EcsS3MockRule rule(String id) {
     EcsS3MockRule rule = TEST_RULE_FOR_MOCK_CLIENT.get();
-    Assert.assertTrue("Test Rule must match id", rule != null && rule.bucket().equals(id));
+    assertThat(rule != null && rule.bucket().equals(id)).as("Test Rule must match id").isTrue();
```
Comment (nastra) [inline dell/src/test/java/org/apache/iceberg/dell/mock/ecs/EcsS3MockRule.java:75]:
```suggestion
    assertThat(rule).isNotNull().extracting(EcsS3MockRule::bucket).isEqualTo(id);
```

File: arrow/src/test/java/org/apache/iceberg/arrow/vectorized/ArrowReaderTest.java [hunk 5]
```diff
@@ -349,20 +348,20 @@ private void readAndCheckHasNextIsIdempotent(
         // Call hasNext() a few extra times.
         // This should not affect the total number of rows read.
         for (int i = 0; i < numExtraCallsToHasNext; i++) {
-          assertTrue(iterator.hasNext());
+          assertThat(iterator.hasNext()).isTrue();
```
Comment (nastra) [inline arrow/src/test/java/org/apache/iceberg/arrow/vectorized/ArrowReaderTest.java:351]:
```suggestion
          assertThat(iterator).hasNext();
```

File: arrow/src/test/java/org/apache/iceberg/arrow/ArrowSchemaUtilTest.java [hunk 6]
```diff
@@ -97,86 +98,86 @@ public void convertComplex() {
...
   }
 
   private void validate(Schema iceberg, org.apache.arrow.vector.types.pojo.Schema arrow) {
-    Assert.assertEquals(iceberg.columns().size(), arrow.getFields().size());
+    assertThat(arrow.getFields().size()).isEqualTo(iceberg.columns().size());
 
     for (Types.NestedField nf : iceberg.columns()) {
       Field field = arrow.findField(nf.name());
-      Assert.assertNotNull("Missing filed: " + nf, field);
+      assertThat(field).as("Missing filed: " + nf).isNotNull();
```
Comment (nastra) [inline arrow/src/test/java/org/apache/iceberg/arrow/ArrowSchemaUtilTest.java:109]:
```suggestion
      assertThat(field).as("Missing field: " + nf).isNotNull();
```

File: arrow/src/test/java/org/apache/iceberg/arrow/ArrowSchemaUtilTest.java [hunk 7]
```diff
@@ -97,86 +98,86 @@ public void convertComplex() {
                 MapType.ofOptional(
                     4, 5, StringType.get(), ListType.ofOptional(6, TimestampType.withoutZone()))));
     org.apache.arrow.vector.types.pojo.Schema arrow = ArrowSchemaUtil.convert(iceberg);
-    Assert.assertEquals(iceberg.columns().size(), arrow.getFields().size());
+    assertThat(arrow.getFields().size()).isEqualTo(iceberg.columns().size());
   }
 
   private void validate(Schema iceberg, org.apache.arrow.vector.types.pojo.Schema arrow) {
-    Assert.assertEquals(iceberg.columns().size(), arrow.getFields().size());
+    assertThat(arrow.getFields().size()).isEqualTo(iceberg.columns().size());
```
Comment (nastra) [inline arrow/src/test/java/org/apache/iceberg/arrow/ArrowSchemaUtilTest.java:105]:
```suggestion
    assertThat(arrow.getFields()).hasSameSizeAs(iceberg.columns());
```

File: arrow/src/test/java/org/apache/iceberg/arrow/ArrowSchemaUtilTest.java [hunk 8]
```diff
@@ -97,86 +98,86 @@ public void convertComplex() {
                 MapType.ofOptional(
                     4, 5, StringType.get(), ListType.ofOptional(6, TimestampType.withoutZone()))));
     org.apache.arrow.vector.types.pojo.Schema arrow = ArrowSchemaUtil.convert(iceberg);
-    Assert.assertEquals(iceberg.columns().size(), arrow.getFields().size());
+    assertThat(arrow.getFields().size()).isEqualTo(iceberg.columns().size());
```
Comment (nastra) [inline arrow/src/test/java/org/apache/iceberg/arrow/ArrowSchemaUtilTest.java:101]:
```suggestion
   assertThat(arrow.getFields()).hasSameSizeAs(iceberg.columns());
```

**Review by nastra — COMMENTED** [review]
I'm ok just switching the assertions now and the `EcsMockRule` to a JUnit5 extension in a separate PR. @DaVincii could you open an issue for this please?

**Comment by DaVincii** [comment]
> #7160

@nastra Created an issue for rule migration 7888

**Review by nastra — COMMENTED** [review]
can you please switch `ArrowReaderTest` to Junit5?

```
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.TemporaryFolder;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
 
 /**
  * Test cases for {@link ArrowReader}.
@@ -128,13 +128,16 @@ public class ArrowReaderTest {
           "decimal",
           "decimal_nullable");
 
-  @Rule public final TemporaryFolder temp = new TemporaryFolder();
-
+  @TempDir private File tempDir;
   private HadoopTables tables;
-
   private String tableLocation;
   private List<GenericRecord> rowsWritten;
 
+  @BeforeEach
+  public void before() {
+    tableLocation = tempDir.toURI().toString();
+  }
+
   /**
    * Read all rows and columns from the table without any filter. The test asserts that the Arrow
    * {@link VectorSchemaRoot} contains the expected schema and expected vector types. Then the test
@@ -667,7 +670,7 @@ public class ArrowReaderTest {
   private void writeTable(boolean constantRecords) throws Exception {
     rowsWritten = Lists.newArrayList();
     tables = new HadoopTables();
-    tableLocation = temp.newFolder("test").toString();
+    tableLocation = tempDir.toURI().toString();
 
     Schema schema =
         new Schema(
@@ -871,7 +874,7 @@ public class ArrowReaderTest {
 
   private DataFile writeParquetFile(Table table, List<GenericRecord> records) throws IOException {
     rowsWritten.addAll(records);
-    File parquetFile = temp.newFile();
+    File parquetFile = File.createTempFile("junit", null, tempDir);
```

File: arrow/src/test/java/org/apache/iceberg/arrow/vectorized/parquet/DecimalVectorUtilTest.java [hunk 9]
```diff
@@ -43,26 +43,26 @@ public void testPadBigEndianBytesNegative() {
     byte[] bytes = bigInt.toByteArray();
     byte[] paddedBytes = DecimalVectorUtil.padBigEndianBytes(bytes, 16);
 
-    assertEquals(16, paddedBytes.length);
+    assertThat(paddedBytes.length).isEqualTo(16);
```
Comment (nastra) [inline arrow/src/test/java/org/apache/iceberg/arrow/vectorized/parquet/DecimalVectorUtilTest.java:46]:
```suggestion
    assertThat(paddedBytes).hasSize(16);
```

**Comment by DaVincii** [comment]
@nastra Updated with the review change.

**Review by reswqa — COMMENTED** [review]
It seems that we have some format violations.

```
Execution failed for task ':iceberg-arrow:spotlessJavaCheck'.
> The following files had format violations:
      src/test/java/org/apache/iceberg/arrow/ArrowSchemaUtilTest.java
          @@ -98,7 +98,7 @@
           ················MapType.ofOptional(
           ····················4,·5,·StringType.get(),·ListType.ofOptional(6,·TimestampType.withoutZone()))));
           ····org.apache.arrow.vector.types.pojo.Schema·arrow·=·ArrowSchemaUtil.convert(iceberg);
          -···assertThat(arrow.getFields()).hasSameSizeAs(iceberg.columns());
          +····assertThat(arrow.getFields()).hasSameSizeAs(iceberg.columns());
           ··}
           
           ··private·void·validate(Schema·iceberg,·org.apache.arrow.vector.types.pojo.Schema·arrow)·{
  Run './gradlew :iceberg-arrow:spotlessApply' to fix these violations.
```

**Comment by nastra** [comment]
@DaVincii can you please run `./gradlew spotlessApply` to fix formatting?

**Comment by DaVincii** [comment]
> @DaVincii can you please run `./gradlew spotlessApply` to fix formatting?

Fixed formatting.

**Comment by DaVincii** [comment]
@nastra I have fixed the formatting changes, this should be good for merging now.

**Comment by nastra** [comment]
LGTM, thanks @DaVincii

**Pull request merged**